chapters/tutorials/index.rst

```diff
@@ -0,0 +1,408 @@
+            flow.index(from_csv(fp, field_resolver={'question': 'text'}))
+
+If you run this, it should finish without errors. You won't see much yet because we are not showing anything after we index. But you should see a new directory created with the downloaded dataset:
+
+.. image:: res/downloaded_dataset.png
+   :width: 600
+
+To actually see something we need to specify how we will display it. For our tutorial we will do so in our browser. Add the following after indexing:
+
+.. code-block:: python
+
+        flow.use_rest_gateway(args.port_expose)
+
+        url_html_path = 'file://' + os.path.abspath(
+            os.path.join(
+                os.path.dirname(os.path.realpath(__file__)), 'static/index.html'
+            )
+        )
+        try:
+            webbrowser.open(url_html_path, new=2)
+        except:
+            pass  # intentional pass, browser support isn't cross-platform
+        finally:
+            default_logger.success(
+                f'You should see a demo page opened in your browser, '
+                f'if not, you may open {url_html_path} manually'
+            )
+
+        if not args.unblock_query_flow:
+            flow.block()
+
+For more information on what the Flow is doing, specially in `f.use_rest_gateway(args.port_expose)` and `f.block()` check our `cookbook <https://github.com/jina-ai/jina/blob/master/.github/2.0/cookbooks/Flow.md>`_
+
+Ok, so it seems that we have plenty of work done already. If you run this you will see a new tab open in your browser, and there you will have a text box ready for you to input some text. However, if you try to enter anything you won't get any results. This is because we are using dummy Executors. Our `MyTransformer` and `MyIndexer` aren't actually doing anything. So far they only print a line when they are called. So we need real `Executors`.
+
+This has been plenty of new information you've learned so far, so we won't go deep into `Executors` today. Instead you can copy-paste the ones we are using for `this example <https://github.com/jina-ai/jina/blob/master/jina/helloworld/chatbot/executors.py>`_, save that `executors.py` file in the same directory where the rest of your code is. The important part to understand is that all Executors' behavior is defined in `executors.py`
+
+To try the `Executors` from the Github repo, just add this before the `download_data` function:
+
+.. code-block:: python
+
+    if __name__ == '__main__':
+        from executors import MyTransformer, MyIndexer
+    else:
+        from .executors import MyTransformer, MyIndexer
+
+And remove the dummy executors we made.
+
+And we are done! If you followed all the steps, now you should have something like this in your browser:
+
+.. image:: res/results.png
+   :width: 600
+
+There are still a lot of concepts to learn. So stay tuned for our next tutorials.
+
+If you have any issues following this tutorial, you can always get support from our [Slack community](https://slack.jina.ai/)
```

Comment:
This link, and the others in the community are formatted for .md not .rst right?